x/oracle/keeper/prices.go

```diff
@@ -187,13 +188,32 @@ func (k Keeper) AppendPriceTR(ctx sdk.Context, tokenID uint64, priceTR types.Pri
 	if expiredRoundID := nextRoundID - agc.GetParamsMaxSizePrices(); expiredRoundID > 0 {
 		store.Delete(types.PricesRoundKey(expiredRoundID))
 	}
-	k.IncreaseNextRoundID(ctx, tokenID)
+	roundID := k.IncreaseNextRoundID(ctx, tokenID)
+
+	// update for native tokens
+	// TODO: set hooks as a genral approach
+	var p types.Params
+	// get params from cache if exists
+	if agc != nil {
+		p = agc.GetParams()
+	} else {
+		p = k.GetParams(ctx)
+	}
+	assetIDs := p.GetAssetIDsFromTokenID(tokenID)
```

Comment:
if assetIDs is nil ?

Reply:
then range will just be skipped in line_203

Reply:
ok,make sure it doesn't affect the logic.